x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/ShardFollowTasksExecutor.java

```diff
@@ -100,84 +79,72 @@ public void validate(ShardFollowTask params, ClusterState clusterState) {
 
-        final ShardId leaderShard = params.getLeaderShardId();
-        final ShardId followerShard = params.getFollowShardId();
-        fetchGlobalCheckpoint(leaderClient, leaderShard, leaderGlobalCheckPoint -> {
-            // TODO: check if both indices have the same history uuid
-            if (leaderGlobalCheckPoint == followGlobalCheckPoint) {
-                logger.debug("{} no write operations to fetch", followerShard);
-                retry(leaderClient, followerClient, task, params, followGlobalCheckPoint, imdVersionChecker);
-            } else {
-                assert followGlobalCheckPoint < leaderGlobalCheckPoint : "followGlobalCheckPoint [" + followGlobalCheckPoint +
-                        "] is not below leaderGlobalCheckPoint [" + leaderGlobalCheckPoint + "]";
-                logger.debug("{} fetching write operations, leaderGlobalCheckPoint={}, followGlobalCheckPoint={}", followerShard,
-                    leaderGlobalCheckPoint, followGlobalCheckPoint);
-                Executor ccrExecutor = threadPool.executor(Ccr.CCR_THREAD_POOL_NAME);
-                Consumer<Exception> handler = e -> {
-                    if (e == null) {
-                        task.updateProcessedGlobalCheckpoint(leaderGlobalCheckPoint);
-                        prepare(leaderClient, followerClient, task, params, leaderGlobalCheckPoint, imdVersionChecker);
-                    } else {
-                        task.markAsFailed(e);
-                    }
-                };
-                ChunksCoordinator coordinator = new ChunksCoordinator(followerClient, leaderClient, ccrExecutor, imdVersionChecker,
-                    params.getMaxChunkSize(), params.getNumConcurrentChunks(), params.getProcessorMaxTranslogBytes(), leaderShard,
-                    followerShard, handler);
-                coordinator.createChucks(followGlobalCheckPoint, leaderGlobalCheckPoint);
-                coordinator.start();
+            @Override
+            protected void updateMapping(LongConsumer handler) {
+                Index leaderIndex = params.getLeaderShardId().getIndex();
+                Index followIndex = params.getFollowShardId().getIndex();
+
+                ClusterStateRequest clusterStateRequest = new ClusterStateRequest();
+                clusterStateRequest.clear();
+                clusterStateRequest.metaData(true);
+                clusterStateRequest.indices(leaderIndex.getName());
+
+                leaderClient.admin().cluster().state(clusterStateRequest, ActionListener.wrap(clusterStateResponse -> {
+                    IndexMetaData indexMetaData = clusterStateResponse.getState().metaData().getIndexSafe(leaderIndex);
+                    assert indexMetaData.getMappings().size() == 1 : "expected exactly one mapping, but got [" +
+                        indexMetaData.getMappings().size() + "]";
+                    MappingMetaData mappingMetaData = indexMetaData.getMappings().iterator().next().value;
```

Comment:
can we validate the index uuid of the index we got?

Reply:
I think the `getIndexSafe(...)` already covers this? It throws an error if the UUIDs don't match.

Reply:
fair enough.